adapters/repos/db/vector/hnsw/neighbor_connections.go

```diff
@@ -153,7 +153,7 @@ func (n *neighborFinderConnector) processRecursively(from uint64, results *prior
 				return err
 			}
 		}
-		if results.Len() >= top && dist < results.Top().Dist {
+		if results.Len() > 0 && results.Len() >= top && dist < results.Top().Dist {
```

Comment:
It wasn't immediately obvious if another clause should be added here for when Len() and top are 0.
If the queue is empty and top is 0, should `id, dist` be added to the queue [`results`] to continue processing, or does that indicate that processing is complete?

Reply:
Hi, I forgot to answer here. First, thanks for your contribution. Second, top indicates how many elements you want to add to the results set. A top value of ten means you want to top ten elements. This being said, a top value or zero means your processing is complete. Check the review below and if you agree, please add the changes and I will accept the PR. If you think the review is not correct, also feel free to speak your mind freely. We could further discus it if you like.

Reply:
Ok, makes sense, I'll update accordingly.

Reply:
@abdelr thanks for the explanation, updated as requested (<= as top is an int, not a uint).